### ISSUE TITLE
Add compatible devices (Enduro 3, Fenix 8, vivoactive 6, etc)

### ISSUE THREAD

**Issue by tsdh** [memen45/SubMusic#101]
My new Enduro 3 just arrived and I'd love to hit the trails with some music on the ears.  Unfortunately, right now the Enduro 3 is not listed as compatible with the current version of SubMusic so I can't install it.

If there's a way that I can test-drive a beta-version and report issues, I'm happy to volunteer.

**Comment by memen45** [comment]
Thanks for reporting! Working on an update and will include support for these devices as well as forerunner 165. Cannot give an ETA yet

**Comment by helmut72** [comment]
Is the technical difference to SubMusic for Plex that big to adapt SubMusic for the new devices? SubMusic for Plex has been working since the new devices were introduced.

**Comment by jm355** [comment]
Seems like the update could be as simple as this change https://github.com/memen45/SubMusic/pull/88/

**Comment by memen45** [comment]
Yes, that is correct, but there are some other pending changes and new features that I haven't been able to test fully (and still need work). Minimizing the technical difference with Submusic for Plex is one of the challenges with this update.

**Comment by helmut72** [comment]
> Seems like the update could be as simple as this change #88

I would be surprised if it is much more. Also display size is the same as Fenix 7/Epix 2.

> that is correct, but there are some other pending changes and new features that I haven't been able to test fully 

Why no intermediate release of the current version with new device names for now? We are happy with current features too. Changelog of Submusic for Plex also doesn't mention much new features, but a recompile with new device names. From Connect IQ site:

```
Version [0.1.23] - 30-08-2024

- add support for enduro3, fenix843mm, fenix847mm, fenix8solar47mm, fenix8solar51mm
- add workaround for Garmin SSL Bug "Unknown Error::0" / "Sync Failed",
```

**Comment by memen45** [comment]
The current technical difference between published versions of SubMusic and SubMusic for Plex is quite large. Most notable the sync core memory management was reorganized completely to allow for much larger size playlists. Also a proxy service was added, to allow for non-https, https with incompatible ciphers, and other various server setups that are not supported directly by the watch software. Furthermore, Jellyfin and Emby support are added in the new update for SubMusic. This is all tested, but the UI implementation needs some last testing/improvement. 

The update to close this difference and add support for the new devices is almost ready, but I have very limited time on hand currently. Going back to the last published version is non-trivial unfortunately. Testing requires the most time, and that is where the new update is at now. I hope to have more time available in the near future.

**Comment by helmut72** [comment]
> Going back to the last published version is non-trivial unfortunately

Btw, where is the 0.2.12 source? Github only goes to 0.2.10.

> Furthermore, Jellyfin and Emby support are added in the new update for SubMusic.

Will this be a standalone app or will Submusic (for Subsonic) be a subscription like the version for Plex? That would also explain the delay instead of just adding the new device names in the existing version.

**Comment by memen45** [comment]
Current features of SubMusic will remain free. 

With regards to the proxy service, I do not think there is a secure way (not in theory, and therefore also not in practice) to provide it without a subscription.

For Emby, I have thought about a separate app, mainly to make connecting to the server easier. This will not have priority for now though. Expect it will be part of the SubMusic update (unless memory limits on watches will force separation into separate apps).

**Comment by helmut72** [comment]
Thanks for clarifying.

**Comment by tdkl** [comment]
Any news regarding a future update?

**Comment by cristianpb** [comment]
I have compiled the app for fenix8 using the connect sdk:

```bash
~/.Garmin/ConnectIQ/Sdks/connectiq-sdk-lin-8.1.0-2025-03-04-7ae1ed1cb/bin/monkeyc -d fenix843mm -f monkey.jungle -o myapp.prg -y private_key.der -w --debug-log-level=3 --typecheck=0
```

I have installed the app by moving the file ` myapp.prg` to the `Garmin/APPS`  folder.

I am able to browser the playlists and but I have some problems when syncing the playlists. The playlists says to be synced, but when I launch the sync process the sync transfer finish abruptly, like if the sync process is skipped.

**Comment by cristianpb** [comment]
> I have compiled the app for fenix8 using the connect sdk:
> 
> ~/.Garmin/ConnectIQ/Sdks/connectiq-sdk-lin-8.1.0-2025-03-04-7ae1ed1cb/bin/monkeyc -d fenix843mm -f monkey.jungle -o myapp.prg -y private_key.der -w --debug-log-level=3 --typecheck=0
> 
> I have installed the app by moving the file ` myapp.prg` to the `Garmin/APPS` folder.
> 
> I am able to browser the playlists and but I have some problems when syncing the playlists. The playlists says to be synced, but when I launch the sync process the sync transfer finish abruptly, like if the sync process is skipped.

I was having an error 402, related with large playlist in subsonic API.

Using a smaller playlist solves the problem ✅ 

I proposed a PR to add support to these devices in the connectiq app.

**Comment by memen45** [comment]
To be added:

- Enduro 3
- Fenix 8
- Vivoactive 6
- Forerunner 970

**Comment by yannlet** [comment]
Could you please also add Venu 4 (45 mm) ?
Thanks !

**Comment by stonie08** [comment]
Adding the Venu 4 on top of the other announced devices would be greatly appreciated. Now tha navidrome added the minimal client setting I‘m waiting eagerly to be finally able to give it a try.
